parametric/build.sbt

```diff
@@ -0,0 +1,55 @@
+import BenchExecDsl._
+
+enablePlugins(BenchExec)
+
+benchmarks ++= Seq(
+  suiteForEncoding("SetAdd", Seq("parametric-specs/SetAdd.tla")),
+  suiteForEncoding("SetAddDel", Seq("parametric-specs/SetAddDel.tla")),
+  suiteForEncoding("SetSndRcv", Seq("parametric-specs/SetSndRcv.tla")),
+  suiteForEncoding("SetSndRcv_NoFullDrop", Seq("parametric-specs/SetSndRcv_NoFullDrop.tla"))
+)
+
+def suiteForEncoding(name: String, specs: Seq[String]) = {
+  val suiteTimeLimit = "1h"
+
+  val defaultMaxLength = 14
+  val maxLength =
+    // We default to the empty string for fallback so that we
+    // can gracefully deal with the case when the environment
+    // variable is not assigned a value
+    sys.env.getOrElse("ENCODING_COMPARISON_MAX_LENGTH", "") match {
+      case "" => defaultMaxLength
+      case i  => i.toInt
+    }
+
+  def checkCmd(encoding: String, length: Int) = {
+    Cmd(
+      s"${encoding}-length:${length}",
+      Opt("check"),
+      Opt("--init", "Init"),
+      Opt("--inv", "Inv"),
+      Opt("--next", "Next"),
+      Opt("--smt-encoding", encoding),
+      Opt("--length", length),
+      Opt("--cinit", s"CInit${length}"),
+    )
+  }
+
+  def runsForEncoding(encoding: String) = {
+    val lengths = 0.to(maxLength, 2)
+    Bench.Runs(
+      s"run-${encoding}",
+      timelimit = suiteTimeLimit,
+      cmds = lengths.map(checkCmd(encoding, _)),
+      tasks = Seq(Tasks(s"task-${name}-${encoding}", specs))
+    )
+  }
+
+  Bench.Suite(
+    name = s"004parametric-${name}",
```

Comment:
what does "parametric" refer to here exactly? Given the content and the readme, seems like maybe this project should be named "encoding"?

Reply:
The name came from the [apalache-tests](https://github.com/informalsystems/apalache-tests/tree/master/parametric) repo. We also make encoding comparisons in the "examples" project added in #54, so I don't think "encoding" by itself is very descriptive. I'm open to suggestions.

Reply:
I guess that if the principle purpose this project's bench marks is to gauge the effectiveness of the encoding, i'd still be inclined to name this "encoding", even if we happen to also vary the encoding in some places in other projects.

re: "parametric", my main question is this: are these specs are testing parametric models in the way that the specs from the apalache-tests repo were? If so, maybe it's a fitting name. If not, maybe we should change it. WDYT? :)

Reply:
From what I can ascertain, it behaves as the parametric config from apalache-tests, but we added additional runs for the arrays-encoding. In principle, I expect every spec in this repo to (eventually) run with both encodings, that is why I don't want to have a project named "encoding".

Reply:
Ok. Makes sense to me. Thanks!